In [89]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import time
from datetime import datetime

In [115]:
target_date_format = "%Y-%m-%d"

In [18]:
# source Texas DSHS
new_cases_travis = pd.read_csv(r"Texas_County_Cases_per_Day_data.csv")

In [117]:
new_cases_travis.dropna()
new_cases_travis["Date"] = time.strftime(target_date_format, time.strptime(new_cases_travis["Date"],'%m/%d/%y'))

TypeError: strptime() argument 0 must be str, not <class 'pandas.core.series.Series'>

In [62]:
# Source OpenTable (yoy = Year over year)
yoy_texas_reservations_sideways = pd.read_csv("YoY_Seated_Diner_Data.csv")

In [63]:
yoy_texas_reservations_sideways.dropna()

Date  18-Feb  19-Feb  20-Feb  21-Feb  22-Feb  23-Feb  24-Feb  \
0  YoY Percent     0.0    -6.0     0.0     2.0     1.0    -3.0     0.0   

   25-Feb  26-Feb  ...  6-Oct  7-Oct  8-Oct  9-Oct  10-Oct  11-Oct  12-Oct  \
0     9.0     0.0  ... -40.87 -41.46 -32.26 -22.17   -7.88  -15.81  -37.19   

   13-Oct  14-Oct  15-Oct  
0  -39.68  -39.02  -33.04  

[1 rows x 242 columns]

In [64]:
yoy_texas_reservations = pd.DataFrame(columns=["Date", "Percent"])

In [65]:
yoy_texas_reservations_sideways.drop("Date", axis=1)

18-Feb  19-Feb  20-Feb  21-Feb  22-Feb  23-Feb  24-Feb  25-Feb  26-Feb  \
0     0.0    -6.0     0.0     2.0     1.0    -3.0     0.0     9.0     0.0   
1     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
2     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
3     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
4     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   

   27-Feb  ...  6-Oct  7-Oct  8-Oct  9-Oct  10-Oct  11-Oct  12-Oct  13-Oct  \
0     0.0  ... -40.87 -41.46 -32.26 -22.17   -7.88  -15.81  -37.19  -39.68   
1     NaN  ...    NaN    NaN    NaN    NaN     NaN     NaN     NaN     NaN   
2     NaN  ...    NaN    NaN    NaN    NaN     NaN     NaN     NaN     NaN   
3     NaN  ...    NaN    NaN    NaN    NaN     NaN     NaN     NaN     NaN   
4     NaN  ...    NaN    NaN    NaN    NaN     NaN     NaN     NaN     NaN   

   14-Oct  15-Oct  
0  -39.02  -33.04  
1     NaN     NaN  
2     NaN     NaN  
3     NaN     NaN  
4     NaN     NaN  

[5 rows x 241 columns]

In [79]:
for date in yoy_texas_reservations_sideways.columns:
    if date != "Date":
        temp = pd.Series([date, yoy_texas_reservations_sideways[date][0]], index = yoy_texas_reservations.columns)
        yoy_texas_reservations = yoy_texas_reservations.append(temp, ignore_index=True)

In [82]:
# Source: https://data.austintexas.gov/Public-Safety/Austin-Code-COVID-19-Complaint-Cases-Dashboard/5bvq-24pm
covid_complaints = pd.read_csv("Austin_Code_COVID-19_Complaint_Cases.csv")

In [86]:
covid_complaints = covid_complaints["OPENDATE"]

In [87]:
# TODO: COUNT THE NUMBER OF COMPLAINTS PER DAY
for complaint in covid_complaints:
    date_time = datetime.strptime(complaint[0], '%b %d %Y %I:%M%p')complaint[0]

0       10/04/2020 12:00:00 AM
1       05/30/2020 12:00:00 AM
2       03/19/2020 12:00:00 AM
3       04/10/2020 12:00:00 AM
4       03/29/2020 12:00:00 AM
                 ...          
7375    08/26/2020 12:00:00 AM
7376    09/10/2020 12:00:00 AM
7377    08/01/2020 12:00:00 AM
7378    06/25/2020 12:00:00 AM
7379    03/26/2020 12:00:00 AM
Name: OPENDATE, Length: 7380, dtype: object

# The following data is all from YouGov (fear of covid, avoiding crowds, wearing masks, support quarantine)
## All data is from March 2 - October 12

In [108]:
covid_fear = pd.read_csv("covid-fear.csv")
covid_fear

DateTime  USA
0   2020-03-02 23:51:40   34
1   2020-03-17 23:51:40   41
2   2020-03-23 23:51:40   52
3   2020-03-30 23:51:40   60
4   2020-04-06 23:51:40   60
5   2020-04-13 23:51:40   59
6   2020-04-20 23:51:40   60
7   2020-04-27 23:51:40   57
8   2020-05-26 23:51:40   55
9   2020-06-01 23:51:40   54
10  2020-06-08 23:51:40   50
11  2020-06-15 23:51:40   55
12  2020-06-22 23:51:40   55
13  2020-07-06 23:51:40   58
14  2020-07-14 23:51:40   58
15  2020-07-20 23:51:40   58
16  2020-07-27 23:51:40   58
17  2020-08-03 23:51:40   57
18  2020-08-07 23:51:40   48
19  2020-08-10 23:51:40   63
20  2020-08-17 23:51:40   59
21  2020-08-21 23:51:40   49
22  2020-08-25 23:51:40   59
23  2020-09-01 23:51:40   57
24  2020-09-08 23:51:40   55
25  2020-09-14 23:51:40   56
26  2020-09-18 23:51:40   46
27  2020-09-21 23:51:40   58
28  2020-09-28 23:51:40   55
29  2020-10-06 23:51:40   60
30  2020-10-12 23:51:40   57

In [109]:
support_quarantine = pd.read_csv("support-quarantine.csv")
support_quarantine

DateTime  USA
0   2020-03-02 23:51:40   58
1   2020-03-17 23:51:40   61
2   2020-03-23 23:51:40   64
3   2020-03-30 23:51:40   71
4   2020-04-06 23:51:40   69
5   2020-04-13 23:51:40   69
6   2020-04-20 23:51:40   66
7   2020-04-27 23:51:40   66
8   2020-05-06 23:51:40   62
9   2020-05-26 23:51:40   56
10  2020-06-01 23:51:40   56
11  2020-06-08 23:51:40   50
12  2020-06-15 23:51:40   56
13  2020-06-22 23:51:40   53
14  2020-07-06 23:51:40   54
15  2020-07-14 23:51:40   57
16  2020-07-20 23:51:40   50
17  2020-07-27 23:51:40   55
18  2020-08-03 23:51:40   52
19  2020-08-10 23:51:40   58
20  2020-08-17 23:51:40   56
21  2020-08-25 23:51:40   53
22  2020-09-01 23:51:40   52
23  2020-09-08 23:51:40   51
24  2020-09-14 23:51:40   55
25  2020-09-21 23:51:40   55
26  2020-09-28 23:51:40   54
27  2020-10-06 23:51:40   55
28  2020-10-12 23:51:40   56

In [110]:
wearing_masks = pd.read_csv("wearing-masks.csv")
wearing_masks

DateTime  USA
0   2020-03-02 23:51:40    7
1   2020-03-17 23:51:40    5
2   2020-03-23 23:51:40   11
3   2020-03-30 23:51:40   17
4   2020-04-06 23:51:40   29
5   2020-04-13 23:51:40   49
6   2020-04-20 23:51:40   58
7   2020-04-27 23:51:40   63
8   2020-05-26 23:51:40   68
9   2020-06-01 23:51:40   69
10  2020-06-08 23:51:40   66
11  2020-06-15 23:51:40   69
12  2020-06-22 23:51:40   71
13  2020-07-06 23:51:40   73
14  2020-07-14 23:51:40   78
15  2020-07-20 23:51:40   74
16  2020-07-27 23:51:40   77
17  2020-08-03 23:51:40   75
18  2020-08-10 23:51:40   81
19  2020-08-17 23:51:40   77
20  2020-08-25 23:51:40   77
21  2020-09-01 23:51:40   79
22  2020-09-08 23:51:40   75
23  2020-09-14 23:51:40   77
24  2020-09-21 23:51:40   76
25  2020-09-28 23:51:40   78
26  2020-10-06 23:51:40   77
27  2020-10-12 23:51:40   79

In [111]:
avoiding_crowds = pd.read_csv("avoid-crowds.csv")
avoiding_crowds

DateTime  USA
0   2020-03-02 23:51:40   24
1   2020-03-17 23:51:40   49
2   2020-03-23 23:51:40   65
3   2020-03-30 23:51:40   77
4   2020-04-06 23:51:40   72
5   2020-04-13 23:51:40   75
6   2020-04-20 23:51:40   73
7   2020-04-27 23:51:40   72
8   2020-05-26 23:51:40   65
9   2020-06-01 23:51:40   65
10  2020-06-08 23:51:40   60
11  2020-06-15 23:51:40   64
12  2020-06-22 23:51:40   63
13  2020-07-06 23:51:40   66
14  2020-07-14 23:51:40   69
15  2020-07-20 23:51:40   65
16  2020-07-27 23:51:40   65
17  2020-08-03 23:51:40   64
18  2020-08-10 23:51:40   70
19  2020-08-17 23:51:40   64
20  2020-08-25 23:51:40   62
21  2020-09-01 23:51:40   62
22  2020-09-08 23:51:40   64
23  2020-09-14 23:51:40   65
24  2020-09-21 23:51:40   63
25  2020-09-28 23:51:40   64
26  2020-10-06 23:51:40   65
27  2020-10-12 23:51:40   65

In [112]:
# From DSHS and Austin - Travis County COVID-19 Dashboard Public Datasets
hospital_data = pd.read_csv("hospital-data.csv")
hospital_data

Date % of hospital capacity for covid  # of hospitalisations  \
0    2020-04-12                             0.9%                     22   
1    2020-04-13                             1.2%                     26   
2    2020-04-14                             1.1%                     28   
3    2020-04-15                             1.3%                     30   
4    2020-04-16                             0.9%                     22   
..          ...                              ...                    ...   
183  2020-10-12                             2.2%                     91   
184  2020-10-13                             2.3%                     96   
185  2020-10-14                             2.4%                     99   
186  2020-10-15                             2.3%                     99   
187  2020-10-16                             2.3%                     99   

    % of hospitalizations for covid  # ICU Beds for covid  
0                              2.1%                    12  
1                              2.5%                    15  
2                              2.4%                    16  
3                              2.9%                    17  
4                              2.3%                    14  
..                              ...                   ...  
183                            2.9%                    32  
184                            3.1%                    34  
185                            3.1%                    34  
186                            3.1%                    39  
187                            3.1%                    38  

[188 rows x 5 columns]

In [113]:
# Case counts: Austin Travis County COVID19 Daily Counts (Public View)
total_cases = pd.read_csv("total-cases.csv")
total_cases

Last_Update  Total_Reported
0     3/13/20 17:00             0.0
1     3/14/20 20:24             0.0
2     3/15/20 20:23             0.0
3     3/16/20 17:00             0.0
4     3/17/20 17:00             0.0
..              ...             ...
212  10/11/20 22:00         31208.0
213  10/12/20 22:19         31341.0
214  10/13/20 22:52             NaN
215  10/14/20 17:31         31554.0
216  10/15/20 22:16         31650.0

[217 rows x 2 columns]

# Data so far: 
1. New cases, 
2. year on year % restaurant reservations, 
3. covid complaints(in-progress), 
4. % fearing covid , 
5. support for quaranting, 
6. % wearing masks, 
7. % avoiding crowds, 
8. % of total hospital capacity dedicated to covid, 
9. /# of hospitalisations for covid, 
10. % of hospitalisations for covid, 
11. /# of ICU beds for covid, 
12. /# cases

# Target Feature: Number of infections for covid (in 2.5 months)